### PR TITLE
Add Dynamic Skybox Support

### PR DESCRIPTION
**Pull request by hooh-hooah**
- Fixed Default LUT is not being saved.
- Fixed Color Saving
- Added support for Dynamic Skybox

### CODE REVIEW

File: AI_Graphics/AI_Graphics/Inspector/LightingInspector.cs [hunk 1]
```diff
@@ -45,32 +45,37 @@ internal static void Draw(LightingSettings lightingSettings, SkyboxManager skybo
```
Comment (ore-) [inline AI_Graphics/AI_Graphics/Inspector/LightingInspector.cs:45]: with many cubemaps, the selection needs to be restricted else it makes the configuration panels really cramped. can you change the cubemapscrollview to something like this?
`cubeMapScrollView = GUILayout.BeginScrollView(cubeMapScrollView, GUILayout.MaxHeight(300));`

**Comment by ore-** [comment]
something's still not right:
* for unity default skybox, just simply saving preset, colour is changed. _if_ this is to do with default LDR LUT, you can't see this is being defaulted in the corresponding PP panel, as nothing is selected. after the initial preset save, subsequent load/save doesn't change colour
* 2 point gradient box continue to change colour on load save.
* also for unity default skybox, I can't see the sun, even with max fov. In unity editor I can see the sun, and adjusting both the sun disk and sun size has an effect.

unless you are ok to release like this and set users' expectation that these 2 are broken and need to be fixed later, then we have to fix them first or remove them from the release (but this gives me no comfort as it hints that something's wrong with the dynamic skybox implementation).

File: AI_Graphics/AI_Graphics/Inspector/LightingInspector.cs (hunk 1 above)
Comment (hooh-hooah) [reply AI_Graphics/AI_Graphics/Inspector/LightingInspector.cs:45]: Fixed

**Comment by hooh-hooah** [comment]
@ore- 

something's still not right:
* for unity default skybox, just simply saving preset, colour is changed. _if_ this is to do with default LDR LUT, you can't see this is being defaulted in the corresponding PP panel, as nothing is selected. after the initial preset save, subsequent load/save doesn't change colour
* 2 point gradient box continue to change colour on load save.

   - Well it should work try again. the color problem was the problem of LUT.

* also for unity default skybox, I can't see the sun, even with max fov. In unity editor I can see the sun, and adjusting both the sun disk and sun size has an effect.

  - The sun direction follows the brightest light in the scene. For default, it is Camera Light. So if you want to see the sun, you need to place bright light on scene or disable Camera Light.

unless you are ok to release like this and set users' expectation that these 2 are broken and need to be fixed later, then we have to fix them first or remove them from the release (but this gives me no comfort as it hints that something's wrong with the dynamic skybox implementation).

**Pull request merged**